dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -473,6 +481,43 @@ public Response getVersion(@PathParam("id") String datasetId, @PathParam("versio
         }));
     }
 
+    @GET
+    @Path("{id}/uningest")
+    public Response listUningestableFiles(@PathParam("id") String datasetId) {
+        return allowCors(response(req -> ok(uningestInfoService.listUningestableFiles(findDatasetOrDie(datasetId)).stream()
```

Comment:
Do we need to secure somehow this endpoint? IMO this should be available at least for users that have view unpublished dataset permission

Reply:
Good point, added a check for the unpublished dataset permission.